src/js/stream_readable.js

```diff
@@ -138,6 +139,76 @@ Readable.prototype.push = function(chunk, encoding) {
 };
 
 
+Readable.prototype.pipe = function(destination, options) {
+  if (!(destination instanceof Writable || isDuplex(destination))) {
```

Comment:
I think it's enough to check if `destination` is readable:

```javascript
if (destination instanceof Readable) {
    throw new TypeError('pipe excepts writable stream argument');
}
```
In this case the `isDuplex` function can be removed.

Reply:
Yes, You are right.